src/eos-crash-metrics.c

```diff
@@ -120,21 +123,40 @@ get_ostree_repo_url (OstreeRepo *repo, const char *origin)
   return url;
 }
 
-static char *
-get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+static gboolean
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot,
+                                  OstreeRepo    *repo,
+                                  char         **commit_out,
+                                  char         **version_out)
 {
   OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
   g_autoptr(GError) error = NULL;
   const char *csum = NULL;
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) commit_metadata = NULL;
+  const char *version = NULL;
 
   if (!deployment)
     {
       g_warning ("OSTree deployment is not currently booted, cannot read state");
-      return NULL;
+      return FALSE;
```

Comment:
Nitpick: Slightly safer to clear both out-variables to `NULL` if returning `FALSE`. Or (less safe but fine) document that their values are undefined if `FALSE` is returned.

Reply:
Addressed by requiring that they are initialized to `NULL` by the caller.